### PR TITLE
Black

### PR DESCRIPTION
**Pull request by jlubcke**
Running black on everything. Great or garbage?

### CODE REVIEW

File: iommi/admin.py [hunk 1]
```diff
@@ -54,8 +54,7 @@
 from iommi.shortcut import with_defaults
 
 app_verbose_name_by_label = {
-    config.label: config.verbose_name.replace('_', ' ')
-    for config in values(django_apps.app_configs)
+    config.label: config.verbose_name.replace('_', ' ') for config in values(django_apps.app_configs)
```
Comment (boxed) [inline iommi/admin.py:57]: This is clearly worse

File: iommi/admin__tests.py [hunk 2]
```diff
@@ -34,13 +34,15 @@
...
-        request=request
+    result = (
+        Admin.list(
+            parts__list_tests_adminunique__columns__foo__bulk__include=True,
+        )
+        .refine_with_params(
+            app_name='tests',
+            model_name='adminunique',
+        )
+        .bind(request=request)
```
Comment (boxed) [inline iommi/admin__tests.py:45]: WAT

This is a strange reformatting imo

File: iommi/experimental/edit_table.py [hunk 3]
```diff
@@ -1,4 +1,3 @@
 raise Exception(
-    'EditTable/EditColumn has moved out of iommi.experimental. '
-    'Update imports and remove the .experimental part.'
+    'EditTable/EditColumn has moved out of iommi.experimental. ' 'Update imports and remove the .experimental part.'
```
Comment (boxed) [inline iommi/experimental/edit_table.py:2]: this is gnarly heh

File: iommi/refinable.py [hunk 4]
```diff
@@ -188,9 +189,12 @@ def refine_done(self, parent=None):
...
             raise TypeError(
-                    f'{result.__class__.__name__} object has no refinable attribute(s): '
-                    + ', '.join(f'"{k}"' for k in sorted(unknown_args)) + '.\n'
-                    + 'Available attributes:\n    ' + available_keys + '\n'
+                f'{result.__class__.__name__} object has no refinable attribute(s): '
+                + ', '.join(f'"{k}"' for k in sorted(unknown_args))
+                + '.\n'
+                + 'Available attributes:\n    '
+                + available_keys
+                + '\n'
```
Comment (boxed) [inline iommi/refinable.py:197]: Clearly worse

**Comment by boxed** [comment]
As normal I'm not sold on black.

**Comment by jlubcke** [comment]
Ok, noted. I'll take it back to the drawing board and see if I can come up with something better.

**Comment by maxalbert** [comment]
@boxed @jlubcke It's obviously up to you as the maintainers to choose the desired code formatting. Just to mention (in case you are not aware yet) that it is always possible to turn black formatting off for specific lines or code blocks using `# fmt: skip` or `# fmt: off/on` (see the [black docs](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html?highlight=fmt%20on%20off#code-style)).

I fully agree that the above examples which @boxed pointed out are worse than before, but I personally have found that in 98% of cases black does a pretty decent job at creating nicely readable and aesthetically structured code. (Some of its choices looked a little odd to me at first but I have really grown to like them over time.) In my personal experience, not having to think about how to manually format something but simply letting PyCharm and black auto-format code files frees up a lot of headspace for me to allow focusing on more important things. Just my 2 cents as an anecdotal data point. Again, totally up to you whether or not you choose to adopt it or something else.

**Comment by boxed** [comment]
Jlubcke is also for it as I understand it. There is a point to black no doubt.

**Pull request merged**